src/components/cylc/analysis/filter.js

```diff
@@ -26,9 +26,16 @@
  * @return {boolean} Boolean determining if task should be displayed
  */
 export function matchTask (task, tasksFilter) {
-  let ret = true
-  if (tasksFilter.name?.trim()) {
-    ret &&= task.name.includes(tasksFilter.name)
+  let ret = false
+  if (tasksFilter.name.length > 0) {
+    for (let i = 0; i < tasksFilter.name.length; i++) {
+      if (tasksFilter.name[i] === task.name) {
+        ret = true
+      }
+    }
```

Comment:
This ought to be a short-cutting loop.
```suggestion
  if (tasksFilter.name?.length) {
    ret = tasksFilter.name.some((name) => task.name === name)
```
`Array.prototype.some()` is like Python's `any()`

Reply:
I'm an old man that doesn't understand these new fangled loop styles

Reply:
In English think of it like `if there is some name in the tasksFilter.name array, that is equal to task.name`